src/document-click.js

```diff
@@ -248,35 +247,49 @@ export function onDocumentElementClick_(
   // Look for the referenced element.
   const hash = tgtLoc.hash.slice(1);
   let elem = null;
+
   if (hash) {
-    elem = ampdoc.getRootNode().getElementById(hash);
-    if (!elem) {
-      // Fallback to anchor[name] if element with id is not found.
-      // Linking to an anchor element with name is obsolete in html5.
-      elem = ampdoc.getRootNode().querySelector(`a[name=${hash}]`);
-    }
+    const escapedHash = escapeCssSelectorIdent(ampdoc.win, hash);
+    elem = (ampdoc.getRootNode().getElementById(hash) ||
+        // Fallback to anchor[name] if element with id is not found.
+        // Linking to an anchor element with name is obsolete in html5.
+        ampdoc.getRootNode().querySelector(`a[name=${escapedHash}]`));
```

Comment:
This CSS selector here also needs `"` I believe. Please add a test for search of an element with name with a space or such.